### PR TITLE
Only write the simple swing store if there were changes

### PR DESCRIPTION
**Pull request by michaelfig**
A minor efficiency when the swing store doesn't change frequently.

### CODE REVIEW

File: packages/swing-store-simple/simpleSwingStore.js [hunk 1]
```diff
@@ -152,7 +167,7 @@ function makeStorageInMemory() {
  * }
  */
 function makeSwingStore(dirPath, forceReset = false) {
-  const { storage, state } = makeStorageInMemory();
+  const { storage, state, isDirty, markClean } = makeStorageInMemory();
```
Comment (FUDCo) [inline packages/swing-store-simple/simpleSwingStore.js:170]: We should not be exposing `isDirty` and `markClean` on this API.  This API is shared with the LMDB store, for which these calls are not relevant, and anyway they are an internal implementation detail that shouldn't be opened for people to fiddle with.

**Review by FUDCo — CHANGES_REQUESTED** [review]
I'm not sure what the motivation for this is.  This only applies to the file store which should only be used in development.  Is there a payoff to optimizing here?

**Comment by michaelfig** [comment]
> I'm not sure what the motivation for this is.

I wanted to use this as a simple way of "pickling" data, where I run a commit on every block, but only add to/delete from the contents every, say, 30 blocks.

> This only applies to the file store which should only be used in development. Is there a payoff to optimizing here?

Just to verify: is the LMDB store format portable between different platforms (endianness, etc)?  If it is, then I can probably use it directly for my "portable string database".

**Comment by FUDCo** [comment]
> I wanted to use this as a simple way of "pickling" data, where I run a commit on every block, but only add to/delete from the contents every, say, 30 blocks.

`swingset-runner` already has a feature somewhat like this.  You might want to check out what I'm doing there to see if you can steal any of it.  It's a bit lower level (snapshot every crank) but still may have some utility for your purpose.

> Just to verify: is the LMDB store format portable between different platforms (endianness, etc)? If it is, then I can probably use it directly for my "portable string database".

Since LMDB's main performance trick is memory mapping files, I'm fairly sure the files are not portable across machines with different endianness.  However, we have a utility (currently living in the `swingset-runner` package but a standalone executable) that dumps a swingstore database into a quasi-readable, common-format text file.

**Pull request closed without merging**